src/test/sighash_tests.cpp

```diff
@@ -121,8 +121,8 @@ void static RandomTransaction(CMutableTransaction &tx, bool fSingle, bool emptyI
         txin.nSequence = (insecure_rand() % 2) ? insecure_rand() : (unsigned int)-1;
     }
     for (int out = 0; out < outs; out++) {
-        tx.vout.push_back(CTxOut());
-        CTxOut &txout = tx.vout.back();
+        tx.addOut(CTxOut());
```

Comment:
not related to this line:
add certificate signature hash test in sighash_test: create a modified version of SignatureHashOld for certificates and verify if new SignatureHash for certs is producing the same hash